src/components/PrimitiveCheckbox/index.js

```diff
@@ -1,27 +1,27 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import { uniqueId } from '../../../../libs/utils';
+import { uniqueId } from '../../libs/utils';
 import Label from './label';
 import StyledContainer from './styled/container';
 import StyledInnerContainer from './styled/innerContainer';
 
-export default class InputCheckbox extends Component {
+export default class PrimitiveCheckbox extends Component {
     constructor(props) {
         super(props);
         this.inputId = uniqueId('input-checkbox');
         this.inputRef = React.createRef();
     }
 
     componentDidMount() {
-        const { indeterminate } = this.props;
-        this.inputRef.current.indeterminate = indeterminate;
+        const { checked } = this.props;
+        this.inputRef.current.indeterminate = checked === 'indeterminate';
     }
 
     componentDidUpdate(prevProps) {
         const { indeterminate: prevIndeterminate } = prevProps;
-        const { indeterminate } = this.props;
-        if (prevIndeterminate !== indeterminate) {
-            this.inputRef.current.indeterminate = indeterminate;
+        const { checked } = this.props;
+        if (prevIndeterminate !== (checked === 'indeterminate')) {
```

Comment:
Since now there is not indeterminate prop now you need to remove:
```const { indeterminate: prevIndeterminate } = prevProps;```
This code es meant to update the `this.inputRef.current.indeterminate` when checked is changed dynamically:
1. when checked = 'indeterminate' change to true or false
2. when checked = true | false change to 'indeterminate'
This is needed for the component to react to a prop change